core/src/main/scala/org/apache/spark/storage/MemoryStore.scala

```diff
@@ -295,9 +296,9 @@ private[spark] class MemoryStore(blockManager: BlockManager, maxMemory: Long)
       // In this case, we should release the memory after we cache the block there.
       // Otherwise, if we return an iterator, we release the memory reserved here
       // later when the task finishes.
-      if (keepUnrolling) {
-        accountingLock.synchronized {
-          val amountToRelease = currentUnrollMemoryForThisThread - previousMemoryReserved
+      accountingLock.synchronized {
```

Comment:
@srowen releasePendingUnrollMemoryForThisThread means: we will reserved the unroll memory reserved for the unroll block, we need to release that part of memory when we  not use any more, like: we put the unroll successful block to memoryStore, or we put unroll unsuccessful block into the disk, or we complete the task.